### PR TITLE
Testing my-enumerables with RSpec

### PR DESCRIPTION
**Pull request by temesghentekeste**
Dear TSE,

We have cloned a repository from a my-enumerable project done in pair programming approach as part of the Microverse Ruby Curriculum. Starting from that project:

1. We created a feature branch called feature/rspec-my_enumerables
2. We initialized RSpec in that branch
3. We tested all enumerable methods in the Enumerable module
4. We tested for all extreme conditions both for truthy and falsy values as much as possible
5. All the tests passed

Kindly request to review our project before we merge it to the master branch and thanks in advance..
Regards,
The team

### CODE REVIEW

File: spec/my_enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    end
+
+    it 'returns the original range once it is done with the block' do
+      expect(range.my_each { |x| x * 2 }).to eql(range)
+    end
+
+    it 'returns the original hash once it is done with the block' do
+      expect(country_codes_hash.my_each { |country| "country - #{country} " }).to eql(country_codes_hash)
+    end
+  end
```
Comment (adaorachi) [inline spec/my_enumerables_spec.rb]: - [ ] You should ensure that the specs for **EACH** method are wrapped in **ONE** `describe` block. In the highlighted code, a `describe` block should be wrapped around the `#my_each` method, then inside, you can have the `context` block which defines the condition the method it tested in and the `it` block shows the behavior of the method at each condition. For example, you should have something similar to this;

 ```
describe '#my_each' do
  context 'when a block is given' do
    it 'returns an the original array' do
      ....
    end
    ....
  end
 ...
end
```

**Note: This DOES NOT ONLY apply to the highlighted code. You should make the changes for all test blocks in this file.**

**Comment by adaorachi** [comment]
### Hi @temesghentekeste and @kiptubei :wave:

I really commend the effort you put into making this project meet the basic requirement of proper Gitflow specification. Your test suite works and runs well, however there are a couple of changes to be done to make your project look professional and hence get approved.

---

### CHANGES REQUIRED :recycle: 

---

- [ ] The README content should be updated accordingly. In the About section, it should include the features implemeted in this project which is testing all the enumerable methods. You should also ensure you add a Testing Game Logic section to the README describing how to run the test in a local machine, the tool used for the TDD and the possible results which is gotten from the test.

- [ ] Kindly ensure that all files not related to the project are not committed but instead added to .gitignore file, e.g - `vscode` file.

**Review by adaorachi — CHANGES_REQUESTED** [review]

**Comment by adaorachi** [comment]
---

### After making the changes requested above, you can request for another code review by submitting a code review form.

---

Should you have any question, please do reach out to me on Slack with the username - **@adaorachi**.

Happy Coding :relaxed: 

**MaryAnn Chukwuka**

File: README.md [hunk 2]
```diff
@@ -50,6 +53,26 @@ This project consist of different ruby enumerable methods that emulate Ruby's in
 
 - run npm install to install the dependencies for the project.
 
+## 🔨 Testing Game Logic
```
Comment (mosesogwo) [inline README.md]: :exclamation: Please change this title to just **Testing**.
There is no game in this project.

**Review by mosesogwo — APPROVED** [review]
**Congratulations! Your project is now approved:+1:**

**_Good luck in your next project:stars:_**


### Status: Approved:heavy_check_mark:

**Pull request merged**